cmake/FindJUnit.cmake

```diff
@@ -1,36 +1,59 @@
 #####################
 ## FindJUnit.cmake
 ## Top contributors (to current version):
-##   Mathias Preiner
+##   Mathias Preiner, Mudathir Mohamed
 ## This file is part of the CVC4 project.
 ## Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
 ## in the top-level source directory and their institutional affiliations.
 ## All rights reserved.  See the file COPYING in the top-level source
 ## directory for licensing information.
 ##
 # Find JUnit
-# JUnit_FOUND - system has JUnit lib
-# JUnit_JAR - JUnit jar file
-# JUnit_JAR_DEPS - JUnit jar dependencies
+# JUnit_FOUND - should be true for testing
+# JUnit_JAR   - absolute path to JUnit5 jar file
 
-find_package(Java REQUIRED)
-include(UseJava)
-find_package(Hamcrest REQUIRED)
+include(deps-helper)
 
-find_jar(JUnit_JAR NAMES junit junit4)
+find_jar(JUnit_JAR junit-platform-console-standalone
+  PATHS
+    ${DEPS_BASE}/share/java
+    $ENV{PATH} $ENV{HOME} $ENV{CLASSPATH} $ENV{JUNIT_HOME}
+  VERSIONS 1.7.1
+)
 
+set(JUnit_FOUND_SYSTEM FALSE)
 if(JUnit_JAR)
-  set(JUnit_JAR_DEPS ${Hamcrest_JAR})
-  # Determine version of JUnit
-  execute_process(
-    COMMAND ${Java_JAVA_EXECUTABLE} -cp ${JUnit_JAR} junit.runner.Version
-    OUTPUT_VARIABLE JUnit_VERSION
-    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  set(JUnit_FOUND_SYSTEM TRUE)
 endif()
 
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(JUnit
-  REQUIRED_VARS JUnit_JAR JUnit_JAR_DEPS
-  VERSION_VAR JUnit_VERSION)
+if(NOT JUnit_FOUND_SYSTEM)
+  include(ExternalProject)
 
-mark_as_advanced(JUnit_JAR JUnit_JAR_DEPS)
+  # Download junit generator jar
+  ExternalProject_Add(
+    JUnit
+    PREFIX ${DEPS_PREFIX}
+    URL https://repo1.maven.org/maven2/org/junit/platform/junit-platform-console-standalone/1.7.1/junit-platform-console-standalone-1.7.1.jar
```

Comment:
It would be good to define a variable with the version. `JUNIT_VERSION` should contain the version that we found/installed (see [GMP](https://github.com/CVC4/CVC4/blob/master/cmake/FindGMP.cmake#L45)).

Reply:
Done